.github/workflows/build-checks.yml

```diff
@@ -116,19 +133,12 @@ jobs:
         working-directory: build
       - name: Set up problem matcher
         uses: ammaraskar/gcc-problem-matcher@0.3.0
-      # Work around https://github.com/actions/runner-images/issues/8659
-      - name: "Remove GCC 13 from runner image (workaround)"
-        shell: bash
-        run: |
-          sudo rm -f /etc/apt/sources.list.d/ubuntu-toolchain-r-ubuntu-test-jammy.list
-          sudo apt-get update
-          sudo apt-get install -y --allow-downgrades libc6=2.35* libc6-dev=2.35* libstdc++6=12.3.0-1ubuntu1~22.04 libgcc-s1=12.3.0-1ubuntu1~22.04
       - name: Build
         # Do not abort on errors and build/check the whole project
         run: cmake --build . -j $(nproc) -- --keep-going
         working-directory: build
         env:
-          CLAZY_CHECKS: level2,container-inside-loop,heap-allocated-small-trivial-type,isempty-vs-count,qhash-with-char-pointer-key,qproperty-type-mismatch,tr-non-literal,no-rule-of-two-soft,no-non-pod-global-static,no-qproperty-without-notify,no-qstring-allocations,no-function-args-by-value,no-copyable-polymorphic,no-ctor-missing-parent-argument,no-missing-qobject-macro,no-rule-of-three,no-returning-void-expression
+          CLAZY_CHECKS: level2,container-inside-loop,heap-allocated-small-trivial-type,isempty-vs-count,qhash-with-char-pointer-key,qproperty-type-mismatch,tr-non-literal,no-rule-of-two-soft,no-non-pod-global-static,no-qproperty-without-notify,no-qstring-allocations,no-function-args-by-value,no-copyable-polymorphic,no-ctor-missing-parent-argument,no-missing-qobject-macro,no-rule-of-three,no-returning-void-expression,no-no-module-include
```

Comment:
Isn't this just a matter of alter include directives: 
https://github.com/KDE/clazy/blob/master/docs/checks/README-no-module-include.md
I would prefer to fix the issue instead of disable the check.

Reply:
I wonder if this is a clang issue as well. 

Is this the issue we are fighting: 

https://github.com/JoergAtGithub/mixxx/actions/runs/11297991113/job/31426009871#step:9:1008

Reply:
Yes, clazy is based on clang. And the old clang version, where clazy 1.11 is based on does not the C++ features used in the Qt6.4 headers. clazy 1.12 is build on base of a recent clang version and can compile this code.

Reply:
Does it build with clang 15? 

I think the GCC headers are not compatible.

https://askubuntu.com/questions/1525087/after-upgrading-ubuntu-to-24-04-lts-clang-does-not-work-properly

Did you try to use the clang libraries with 
`-stdlib=libc++"`

Reply:
Clazy passes now with unmodified ruleset